schema/table.go

```diff
@@ -182,14 +182,17 @@ func NewTableFromArrowSchema(sc *arrow.Schema) (*Table, error) {
 	for i, field := range fields {
 		columns[i] = NewColumnFromArrowField(field)
 	}
+
+	var permissionsNeededArr []string
+	_ = json.Unmarshal([]byte(permissionsNeeded), &permissionsNeededArr)
```

Comment:
Don't think `Marshal`, `Unmarshal` can fail for `[]string` and if it does I don't think it should panic for a doc related feature

Reply:
I mean it can fail if the string is not a JSON array but we're the ones setting it

Reply:
Can it fail if it's not set?

Reply:
Is that a good test case?

Reply:
If it's not set `nil` array gets serialized to `null` I'll add a test for it

Reply:
Added https://github.com/cloudquery/plugin-sdk/pull/1835/commits/baf5f303c65d2f9680c0641248581cacc78c3144

`nil` columns are always recreated as non `nil` due to https://github.com/cloudquery/plugin-sdk/blob/341c54641fc2de65c23287276989d9aebca2aecd/schema/table.go#L181